rfcs/0034-system-parameters.md

```diff
@@ -0,0 +1,115 @@
+---
+feature: system-parameters
+authors:
+  - "Eric Fu"
+start_date: "2023/01/11"
+---
+
+# RFC: System Parameters on Meta
+
+## Motivation
+
+Currently, we have 2 kinds of configurations:
+
+- Command-line arguments
+- Configuration file `risingwave.toml`
+
+[Distinguish system config variables and cli options #5676](https://github.com/risingwavelabs/risingwave/issues/5676) clarifies the difference between CLI args and config file:
+
+> Intuitively, *Options* should contain basic user-facing options, e.g., addr & host and maybe some feature gates, while *Config* contains deep-dark internal system variables to fine-tune the system performance.
+
+However, this is not a complete solution because
+
+1. **`risingwave.toml` is a local file**, which makes it possible that multiple nodes could see different parameters and this is absolutely unacceptable. For example, if some nodes use `data_directory = 'hummock_001'` and others use `data_directory = 'hummock_002'`, the stored data will be corrupted.
+2. **Unable to `SET` a config after the process started**. Even if changes are written to disk, new nodes still miss these changes.
+
+This proposal will continue to improve the configuration mechanism but also repudiate the design of #5676.
+
+Previously I think this might not be an urgent issue until I realized that our Cloud platform couples tightly with more and more configurations. The later we commence, the heavier burden it will become. 
+
+## Design
+
```

Comment:
To summarize, I think it's two designs:
- Distinguish system (cluster) configs from process configs. This is to resolve the main concerns in the motivation section.
- Introduce layered config to replace separated configs. (e.g., CLI args become subset of config file. Instead of two distinct set of configs.) Although LGTM, it doesn't seem to belong to any problem described in motivation 🤪

Reply:
I'll improve that part later.